lxd/util/sys.go

```diff
@@ -110,3 +110,14 @@ func AddFileDescriptor(fdFiles *[]*os.File, file *os.File) int {
 	*fdFiles = append(*fdFiles, file)
 	return 2 + len(*fdFiles) // Use 2+fdFiles count, as first user file descriptor is 3.
 }
+
+// ShortenedSocketPath creates a shorter alternative path to a socket by using the file descriptor to the directory of the socket file.
+// Used to handle paths > 108 chars.
+func ShortenedSocketPath(originalSockPath string, fdFiles *[]*os.File) (string, error) {
+	configSockDir, err := os.Open(filepath.Dir(originalSockPath)) // Files opened here must be closed outside this function once they are not needed anymore.
+	if err != nil {
+		return "", fmt.Errorf("Couldn't open device socket directory %q: %w", filepath.Dir(originalSockPath), err)
+	}
+
+	return fmt.Sprintf("/proc/self/fd/%d/%s", AddFileDescriptor(fdFiles, configSockDir), filepath.Base(originalSockPath)), nil
```

Comment:
I feel this would warrant a comment as to why you are opening the directory of the file path rather than the file path itself and then passing back a path to the opened directory's fd with the original file inside it.